api/indexer/client_test.go

```diff
@@ -9,43 +9,54 @@ import (
 	"testing"
 	"time"
 
+	"github.com/ava-labs/avalanchego/database"
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/trace"
 	"github.com/stretchr/testify/require"
 
 	"github.com/ava-labs/hypersdk/api"
+	"github.com/ava-labs/hypersdk/chain"
 	"github.com/ava-labs/hypersdk/chain/chaintest"
+	"github.com/ava-labs/hypersdk/codec"
 )
 
 func TestIndexerClientBlocks(t *testing.T) {
 	const (
-		numExecutedBlocks = 4
-		blockWindow       = 2
-		numTxs            = 3
+		blockWindow = 2
+		numTxs      = 3
 	)
 
 	testCases := []struct {
-		name      string
-		blkHeight uint64
-		err       error
+		name              string
+		blkHeight         uint64
+		blkHeightErr      error
+		numExecutedBlocks int
 	}{
 		{
-			name:      "success",
-			blkHeight: numExecutedBlocks - 1,
-			err:       nil,
+			name:              "success",
+			blkHeight:         3,
+			blkHeightErr:      nil,
+			numExecutedBlocks: 4,
+		},
+		{
+			name:              "missing block",
+			blkHeight:         0,
+			blkHeightErr:      errBlockNotFound,
+			numExecutedBlocks: 4,
 		},
 		{
-			name:      "missing block",
-			blkHeight: 0,
-			err:       errBlockNotFound,
+			name:              "no blocks",
+			blkHeight:         0,
```

Comment:
In production, we should be able to serve the genesis block, should we generate a sequence of blocks that includes a genesis block?